Sources/Logging/Logging.swift

```diff
@@ -1120,39 +1283,68 @@ public struct StreamLogHandler: LogHandler {
     }
 
     // internal for testing only
-    internal init(label: String, stream: _SendableTextOutputStream) {
+    internal init(label: String, metadataProvider: Logger.MetadataProvider, stream: _SendableTextOutputStream) {
         self.label = label
+        self.metadataProvider = metadataProvider
         self.stream = stream
     }
 
     public func log(level: Logger.Level,
                     message: Logger.Message,
-                    metadata: Logger.Metadata?,
+                    metadata explicitMetadata: Logger.Metadata?,
                     source: String,
                     file: String,
                     function: String,
                     line: UInt) {
-        let prettyMetadata = metadata?.isEmpty ?? true
-            ? self.prettyMetadata
-            : self.prettify(self.metadata.merging(metadata!, uniquingKeysWith: { _, new in new }))
+        let effectiveMetadata = StreamLogHandler.prepareMetadata(base: self.metadata, provider: self.metadataProvider, explicit: explicitMetadata)
+
+        let prettyMetadata: String?
```

Comment:
`String!`?

Reply:
Oh good catch, just `String` perhaps

Reply:
The optional actually works out well here since the self.prettyMetadata may be optional so this is a nice way to handle it. Keeping as is